cells/klayout/pymacros/cells/draw_mos.py

```diff
@@ -578,7 +578,7 @@ def draw_pmos(layout, l, w, ld, nf, grw, bulk, volt, deepnwell, pcmpgr):
                 pya.Trans.new(pya.Point.new(((l + ld + ld_violat + cont2ply - cmp2cont) * i + 2 * (ld + ld_violat) + l - cont_size - dx), dy)),
                 pya.Point.new(-(cont2cont+cont_size), 0), pya.Point.new(0, (cont2cont+cont_size)), nx, ny))
             # Metals
-            cell.shapes(metal1).insert(pya.Box((ld + 2 * ld_violat + l + cont2ply - cmp2cont)*(i + 1), - metal_violat,
+            cell.shapes(metal1).insert(pya.Box((ld + 2 * ld_violat + l + cont2ply - cmp2cont)*(i + 1)- metal_violat, - metal_violat,
```

Comment:
This line is way to verbose, and very hard to tell what it's doing. We as explicit an order of operations as possible. In general the long chain of object creation and transformation should be broken up it's very unclear to me what's happening.

```python
metal1_box_x = (ld + 2 * ld_violat + l + cont2ply - cmp2cont)*(i + 1) - metal_violat
metal1_box_y = -metal_violat
metal1_box_width = (ld + metal_violat + ld_violat) + (ld + ld_violat + l + cont2ply - cmp2cont)*(i + 1) - (cont_size-2*cmp2cont)
metal1_box_height =  w + metal_violat

pya.Box(metal1_box_x, metal1_box_y, metal1_box_width, metal1_box_height)

```